An integrated genomics approach deciphering human genome codes shaping HIV-1 proviral transcription and fate

Holly Ruess<sup>1</sup>, Jeon Lee<sup>1</sup>, Carlos Guzman<sup>2,3</sup>, Venkat Malladi<sup>1</sup>, and Iván D’Orso<sup>2,4</sup>

<sup>1</sup>Bioinformatics Core Facility, The University of Texas Southwestern Medical Center, UT Southwestern, Dallas, TX 75390, USA 
<sup>2</sup>Department of Microbiology, The University of Texas Southwestern Medical Center, UT Southwestern, Dallas, TX 75390, USA
<sup>3</sup>Present address: Bioinformatics and Systems Biology Graduate Program, University of California San Diego, La Jolla, CA 92093
<sup>4</sup>Lead contact: Ivan.Dorso@utsouthwestern.edu


SUMMARY 
A large body of work has revealed fundamental principles of HIV-1 integration into the human genome. However, the effect of the integration site to proviral transcription activity has so far remained elusive. Here we combine open-source, large-scale datasets including epigenetics, transcriptome, and 3D genome architecture to interrogate the chromatin states, transcription activity landscape, and nuclear sub-compartments around HIV-1 integration sites in CD4+ T cells to decipher human genome codes shaping the transcription of proviruses groups defined based on their position and orientation in the human genome. Using a Hidden Markov Model, we describe the importance of specific chromatin states and genome architecture in the control of HIV-1 transcription activity. Additionally, a machine-learning logistic regression model reveals upstream chromatin accessibility, transcription activity, and categorical nuclear sub-compartments as optimal features predicting HIV-1 transcriptional outcomes. We finally demonstrate clinical relevance by interrogating the positions of clonally expanded, intact proviruses persisting under suppressive therapy in patient samples containing and provide a compass compatible with clinical decision-making.
