# An integrated genomics approach deciphering human genome codes shaping HIV-1 proviral transcription and fate<br />

Holly Ruess<sup>1</sup>, Jeon Lee<sup>1</sup>, Carlos Guzman<sup>2,3</sup>, Venkat Malladi<sup>1</sup>, and Iván D’Orso<sup>2,4</sup><br />

<sup>1</sup>Lyda Hill Department of Bioinformatics, The University of Texas Southwestern Medical Center, UT Southwestern, Dallas, TX 75390, USA <br />
<sup>2</sup>Department of Microbiology, The University of Texas Southwestern Medical Center, UT Southwestern, Dallas, TX 75390, USA<br />
<sup>3</sup>Present address: Bioinformatics and Systems Biology Graduate Program, University of California San Diego, La Jolla, CA 92093<br />
<sup>4</sup>Lead contact: Ivan.Dorso@utsouthwestern.edu<br />


SUMMARY<br />
A large body of work has revealed fundamental principles of HIV integration into the human genome, but the effect of the integration site to viral transcription has so far remained elusive. Tackling this problem requires the analysis of multiple genomic datasets for thousands of proviral integration sites. Here, we generated and combined large-scale datasets including epigenetics, transcriptome, and three-dimensional genome architecture to interrogate the chromatin states, transcription activity, and nuclear sub-compartments around HIV integrations in Jurkat CD4+ T cells to decipher human genome codes shaping HIV transcription. Using a Hidden Markov Model and ranked informative values prior to a Machine-learning logistic regression model, we defined nuclear sub-compartments and chromatin states contributing to genomic architecture, transcriptional activity, and nucleosome density of regions neighboring the integration site, as additive features influencing HIV expression. Our integrated genomics approach also allows for a robust experimental design, in which HIV can be genetically introduced into precise genomic locations with known regulatory features to assess the relationship of integration positions to viral transcription and fate.
